[rocThrust] Update host and device system CMake options (#4983)
ROCm/rocm-libraries: projects/rocthrust/docs/install/rocThrust-install-with-cmake.rst

* ``USE_HIPCXX``. Set this to ``ON`` to build with CMake HIP language support. Setting this to ``ON`` eliminates the need to use ``CXX=hipcc``. Default is ``OFF``.
* ``ROCPRIM_FETCH_METHOD`` and  ``ROCRAND_FETCH_METHOD``. Set these to the method to use to download the rocPRIM and rocRAND components, respectively. Can be set to ``PACKAGE``, ``DOWNLOAD``, or ``MONOREPO``. Set to ``MONOREPO`` if the component isn't already installed and you're building rocThrust from within a clone of the `rocm-libraries <https://github.com/ROCm/rocm-libraries/>`_ repository that includes the component. Set to ``DOWNLOAD`` if the component isn't installed and you aren't in a clone of the ``rocm-libraries`` repository that includes the component. ``DOWNLOAD`` will clone the repository using sparse checkout so that only the necessary files are downloaded. Set to ``PACKAGE`` if the component is already installed. If the component isn't installed, it'll be downloaded form the repository in the same way as using the ``DOWNLOAD`` option. The default method is ``PACKAGE``.

> * ``LINK_HIP_DEVICE_LIBS``. Set to ``OFF`` to prevent linking against device dependencies (including rocPRIM). In this case, the build can only be used for host-side computation. Default is ``ON``.

Reviewer: ```suggestion
* ``LINK_HIP_DEVICE_LIBS``. Set to ``OFF`` to prevent linking against device dependencies, including rocPRIM, and the build can only be used for host-side computation. Default is ``ON``.
```

Author: Fixed, thanks.

> 

Reviewer: Do the two new options, THRUST_HOST_SYSTEMS and THRUST_DEVICE_SYSTEMS, need to be added to the list, or were they omitted intentionally?

If they were omitted intentionally, we should take them out of the changelog.

Author: I overlooked these - I've added them into the latest commit. Thanks!

* ``USE_HIPCXX``. Set this to ``ON`` to build with CMake HIP language support. Setting this to ``ON`` eliminates the need to use ``CXX=hipcc``. Default is ``OFF``.
* ``ROCPRIM_FETCH_METHOD`` and  ``ROCRAND_FETCH_METHOD``. Set these to the method to use to download the rocPRIM and rocRAND components, respectively. Can be set to ``PACKAGE``, ``DOWNLOAD``, or ``MONOREPO``. Set to ``MONOREPO`` if the component isn't already installed and you're building rocThrust from within a clone of the `rocm-libraries <https://github.com/ROCm/rocm-libraries/>`_ repository that includes the component. Set to ``DOWNLOAD`` if the component isn't installed and you aren't in a clone of the ``rocm-libraries`` repository that includes the component. ``DOWNLOAD`` will clone the repository using sparse checkout so that only the necessary files are downloaded. Set to ``PACKAGE`` if the component is already installed. If the component isn't installed, it'll be downloaded form the repository in the same way as using the ``DOWNLOAD`` option. The default method is ``PACKAGE``.

> * ``LINK_HIP_DEVICE_LIBS``. Set to ``OFF`` to prevent linking against device dependencies, including rocPRIM, and the build can only be used for host-side computation. Default is ``ON``.

Reviewer 2: LGTM.  I would just make it more explicit somewhere here that when THRUST_DEVICE_SYSTEM is set to something other than HIP, you must turn off LINK_HIP_DEVICE_LIBS.

Author: Good idea, thanks. I've updated the documentation to include that information now.
